actionview/lib/action_view/renderer/partial_renderer.rb

```diff
@@ -481,7 +481,13 @@ def partial_path(object = @object)
           raise ArgumentError.new("'#{object.inspect}' is not an ActiveModel-compatible object. It must implement :to_partial_path.")
         end
 
-        if @view.prefix_partial_path_with_controller_namespace
+        should_prefix = if object.respond_to?(:prefix_partial_path_with_controller_namespace)
+          object.prefix_partial_path_with_controller_namespace
+        else
+          @view.prefix_partial_path_with_controller_namespace
+        end
```

Comment:
What do you think about moving this to a method on it's own? Looking at this code I find it difficult to understand what it's doing since both are providing a prefix.